### PR TITLE
refactor: minor refactoring of cluster connection function

### LINKED ISSUE #7333: Add missing error check for verifyClusterConnection
### Summary 
There is a missing error check for this function. A bit of refactoring is needed.

### PR DESCRIPTION
**Pull request by ankitm123**
Signed-off-by: ankitm123 <ankitmohapatra123@gmail.com>

#### Submitter checklist

- [x] Change is code complete and matches issue description.
- [x] Change is covered by existing or new tests.

#### Description
Just a minor refactor to improve error handling

#### Special notes for the reviewer(s)
Another observation is we use `BootOptions`, since the package name is boot, it can be simplified to Options instead to avoid stutter. Thoughts? 

#### Which issue this PR fixes

fixes #7333

### CODE REVIEW

**Comment by ankitm123** [comment]
any idea why codegen fails?

File: pkg/cmd/boot/boot.go [hunk 1]
```diff
@@ -568,13 +568,17 @@ func (o *BootOptions) verifyRequirements(requirements *config.RequirementsConfig
 
 func (o *BootOptions) verifyClusterConnection() error {
 	client, ns, err := o.KubeClientAndNamespace()
-	if err == nil {
-		_, err = client.CoreV1().Pods(ns).List(metav1.ListOptions{})
```
Comment (hferentschik) [inline pkg/cmd/boot/boot.go:572]: good catch, in the diff I initially did not see the missing error check listing the pods ;-)

**Comment by hferentschik** [comment]
@ankitm123, here is the error:

```
go: downloading github.com/go-logr/logr v0.2.0
go: extracting github.com/go-logr/logr v0.2.0
# k8s.io/klog/v2
/builder/home/.**/codegen/go/pkg/mod/k8s.io/klog/v2@v2.1.0/klog.go:690:45: undefined: logr.InfoLogger
/builder/home/.**/codegen/go/pkg/mod/k8s.io/klog/v2@v2.1.0/klog.go:702:43: undefined: logr.InfoLogger
/builder/home/.**/codegen/go/pkg/mod/k8s.io/klog/v2@v2.1.0/klog.go:706:48: undefined: logr.InfoLogger
/builder/home/.**/codegen/go/pkg/mod/k8s.io/klog/v2@v2.1.0/klog.go:721:44: undefined: logr.InfoLogger
/builder/home/.**/codegen/go/pkg/mod/k8s.io/klog/v2@v2.1.0/klog.go:739:55: undefined: logr.InfoLogger
/builder/home/.**/codegen/go/pkg/mod/k8s.io/klog/v2@v2.1.0/klog.go:764:32: undefined: logr.InfoLogger
/builder/home/.**/codegen/go/pkg/mod/k8s.io/klog/v2@v2.1.0/klog.go:881:43: undefined: logr.InfoLogger
/builder/home/.**/codegen/go/pkg/mod/k8s.io/klog/v2@v2.1.0/klog.go:1234:10: undefined: logr.InfoLogger'
make: *** [Makefile.codegen:59: codegen-openapi] Error 1
FAILED make verify-generation-complete
```

A bit unclear to me what's going on.

**Comment by hferentschik** [comment]
same error actually as in pull request #7328

**Comment by hferentschik** [comment]
```
/builder/home/.**/codegen/go/pkg/mod/k8s.io/klog/v2@v2.1.0/klog.go:690:45: undefined: logr.InfoLogger
/builder/home/.**/codegen/go/pkg/mod/k8s.io/klog/v2@v2.1.0/klog.go:702:43: undefined: logr.InfoLogger
/builder/home/.**/codegen/go/pkg/mod/k8s.io/klog/v2@v2.1.0/klog.go:706:48: undefined: logr.InfoLogger
/builder/home/.**/codegen/go/pkg/mod/k8s.io/klog/v2@v2.1.0/klog.go:721:44: undefined: logr.InfoLogger
/builder/home/.**/codegen/go/pkg/mod/k8s.io/klog/v2@v2.1.0/klog.go:739:55: undefined: logr.InfoLogger
/builder/home/.**/codegen/go/pkg/mod/k8s.io/klog/v2@v2.1.0/klog.go:764:32: undefined: logr.InfoLogger
/builder/home/.**/codegen/go/pkg/mod/k8s.io/klog/v2@v2.1.0/klog.go:881:43: undefined: logr.InfoLogger
/builder/home/.**/codegen/go/pkg/mod/k8s.io/klog/v2@v2.1.0/klog.go:1234:10: undefined: logr.InfoLogger: failed to run 'go get -u k8s.io/kube-openapi/cmd/openapi-gen@36ebc4887cdc' command in directory '', output: 'go: finding k8s.io 36ebc4887cd
```

**Comment by hferentschik** [comment]
Hmm, https://github.com/kubernetes/klog/issues/163. Something seems broken upstream somewhere.

**Comment by hferentschik** [comment]
So, locally I see the same error:
```
$make generate-openapi
Generating OpenAPI structs for Kubernetes Clients
./build/codegen --generator-version 36ebc4887cdc openapi --output-package=pkg/client --input-package=github.com/jenkins-x/jx/v2/pkg/apis --group-with-version=jenkins.io:v1 --version=2.0 --module-name=github.com/jenkins-x/jx/v2 --semver=v2
INFO[0000] installing k8s.io/kube-openapi/cmd/openapi-gen with version 36ebc4887cdc via 'go get' to /Users/hardy/.jx/codegen/go  app=codegen
error installing kubernetes openapi tools: error running go get -u k8s.io/kube-openapi/cmd/openapi-gen@36ebc4887cdc, output go: finding k8s.io 36ebc4887cdc
go: finding k8s.io/kube-openapi 36ebc4887cdc
go: finding k8s.io/kube-openapi/cmd/openapi-gen 36ebc4887cdc
go: finding k8s.io/kube-openapi/cmd 36ebc4887cdc
go: finding github.com/modern-go/concurrent latest
go: finding k8s.io/gengo latest
go: finding github.com/PuerkitoBio/urlesc latest
go: finding golang.org/x/net latest
go: finding golang.org/x/tools latest
go: finding golang.org/x/xerrors latest
# k8s.io/klog/v2
/Users/hardy/.jx/codegen/go/pkg/mod/k8s.io/klog/v2@v2.1.0/klog.go:690:45: undefined: logr.InfoLogger
/Users/hardy/.jx/codegen/go/pkg/mod/k8s.io/klog/v2@v2.1.0/klog.go:702:43: undefined: logr.InfoLogger
/Users/hardy/.jx/codegen/go/pkg/mod/k8s.io/klog/v2@v2.1.0/klog.go:706:48: undefined: logr.InfoLogger
/Users/hardy/.jx/codegen/go/pkg/mod/k8s.io/klog/v2@v2.1.0/klog.go:721:44: undefined: logr.InfoLogger
/Users/hardy/.jx/codegen/go/pkg/mod/k8s.io/klog/v2@v2.1.0/klog.go:739:55: undefined: logr.InfoLogger
/Users/hardy/.jx/codegen/go/pkg/mod/k8s.io/klog/v2@v2.1.0/klog.go:764:32: undefined: logr.InfoLogger
/Users/hardy/.jx/codegen/go/pkg/mod/k8s.io/klog/v2@v2.1.0/klog.go:881:43: undefined: logr.InfoLogger
/Users/hardy/.jx/codegen/go/pkg/mod/k8s.io/klog/v2@v2.1.0/klog.go:1234:10: undefined: logr.InfoLogger: failed to run 'go get -u k8s.io/kube-openapi/cmd/openapi-gen@36ebc4887cdc' command in directory '', output: 'go: finding k8s.io 36ebc4887cdc
go: finding k8s.io/kube-openapi 36ebc4887cdc
go: finding k8s.io/kube-openapi/cmd/openapi-gen 36ebc4887cdc
go: finding k8s.io/kube-openapi/cmd 36ebc4887cdc
go: finding github.com/modern-go/concurrent latest
go: finding k8s.io/gengo latest
go: finding github.com/PuerkitoBio/urlesc latest
go: finding golang.org/x/net latest
go: finding golang.org/x/tools latest
go: finding golang.org/x/xerrors latest
# k8s.io/klog/v2
/Users/hardy/.jx/codegen/go/pkg/mod/k8s.io/klog/v2@v2.1.0/klog.go:690:45: undefined: logr.InfoLogger
/Users/hardy/.jx/codegen/go/pkg/mod/k8s.io/klog/v2@v2.1.0/klog.go:702:43: undefined: logr.InfoLogger
/Users/hardy/.jx/codegen/go/pkg/mod/k8s.io/klog/v2@v2.1.0/klog.go:706:48: undefined: logr.InfoLogger
/Users/hardy/.jx/codegen/go/pkg/mod/k8s.io/klog/v2@v2.1.0/klog.go:721:44: undefined: logr.InfoLogger
/Users/hardy/.jx/codegen/go/pkg/mod/k8s.io/klog/v2@v2.1.0/klog.go:739:55: undefined: logr.InfoLogger
/Users/hardy/.jx/codegen/go/pkg/mod/k8s.io/klog/v2@v2.1.0/klog.go:764:32: undefined: logr.InfoLogger
/Users/hardy/.jx/codegen/go/pkg/mod/k8s.io/klog/v2@v2.1.0/klog.go:881:43: undefined: logr.InfoLogger
/Users/hardy/.jx/codegen/go/pkg/mod/k8s.io/klog/v2@v2.1.0/klog.go:1234:10: undefined: logr.InfoLogger'
make: *** [codegen-openapi] Error 1
```

**Comment by hferentschik** [comment]
I opened issue #7338

**Pull request merged**